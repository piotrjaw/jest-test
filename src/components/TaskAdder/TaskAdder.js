import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class TaskAdder extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <div />
    );
  }
}
