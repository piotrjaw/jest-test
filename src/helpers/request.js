import axios from 'axios';

const contructUrl = (url) => `http://localhost:5000${url}`;

export const del = (url) => axios.delete(contructUrl(url));

export const get = (url) => axios.get(contructUrl(url));

export const post = (url, data) => axios.post(contructUrl(url), data);

export const put = (url, data) => axios.put(contructUrl(url), data);
