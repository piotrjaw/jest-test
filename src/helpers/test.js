import React from 'react';

export const getWrapper = ({ Component, method, props }) => {
  return method(
    <Component
      { ...props }
    >
      { props.children }
    </Component>
  );
};

export const getMockStore = (mockFn, state = {}) => ({
  subscribe: mockFn(),
  dispatch: mockFn(),
  getState: mockFn(() => ({
    ...state,
  })),
});
