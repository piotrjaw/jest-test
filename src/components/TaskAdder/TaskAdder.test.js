import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { getWrapper } from '../../helpers/test';
import TaskAdder from './TaskAdder';

const defaultProps = {
};

const setupWrapper = (config = {}) => {
  const { method = shallow, props = {} } = config;
  return getWrapper({
    Component: TaskAdder,
    method,
    props: {
      ...defaultProps,
      ...props
    }
  });
};

let wrapper;

beforeEach(() => {
  wrapper = setupWrapper();
});

afterEach(() => {
  expect(toJSON(wrapper.render())).toMatchSnapshot();
  wrapper = null;
  jest.resetAllMocks();
});

describe('TaskAdder component', () => {
  it('should render', () => {
  });

  it('should handle input change', () => {
  });

  it.skip('should handle button click', () => {
  });
});
