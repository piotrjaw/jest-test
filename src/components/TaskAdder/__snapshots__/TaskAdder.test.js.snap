// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskAdder component should handle input change 1`] = `<div />`;

exports[`TaskAdder component should render 1`] = `<div />`;
